extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java

```diff
@@ -278,6 +227,106 @@ private void consumeService(Class<?> serviceClass, BiConsumer<Class<?>, Collecti
         }
     }
 
+    private void resolveLiquibaseResources(
+            BuildProducer<ServiceProviderBuildItem> services,
+            BuildProducer<ReflectiveClassBuildItem> reflective,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized,
+            BuildProducer<NativeImageResourceBuildItem> resource) {
+
+        var jarUrlTemplate = "jar:%s!/";
+        try {
+            var srcUrl = Objects.requireNonNull(Liquibase.class
+                    .getProtectionDomain()
+                    .getCodeSource()
+                    .getLocation());
+            var jarUrl = new java.net.URL(jarUrlTemplate.formatted(srcUrl.toString())).toURI();
+
+            try (var ignored = FileSystems.newFileSystem(jarUrl, Collections.EMPTY_MAP)) {
+                var rootPath = Paths.get(jarUrl);
+                loadLiquibaseServiceProviderConfig(rootPath, services, reflective, runtimeInitialized);
+                loadLiquibaseRootProperties(rootPath, resource);
+                loadLiquibaseXsdResources(rootPath, resource);
+            }
+        } catch (URISyntaxException | IOException ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    private void loadLiquibaseServiceProviderConfig(
+            Path path,
+            BuildProducer<ServiceProviderBuildItem> services,
+            BuildProducer<ReflectiveClassBuildItem> reflective,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized) throws IOException {
+
+        List<Class<?>> runtimeInitializationServices = List.of(
+                liquibase.configuration.ConfigurationValueProvider.class);
+        Files.list(path.resolve("META-INF/services"))
+                .map(Path::getFileName)
+                .map(Path::toString)
+                .map(this::classFromString)
+                .filter(not(runtimeInitializationServices::contains))
```

Comment:
I think I would avoid transforming this to a class then a String again. You can make `runtimeInitializationServices` a set of strings and use `liquibase.configuration.ConfigurationValueProvider.class`.

Reply:
Done.
I realised the same happened for the static initialisation. I changed it there, too.